data/src/main/java/me/androidbox/data/worker_manager/TaskCreateSynchronizerWorker.kt

```diff
@@ -0,0 +1,78 @@
+package me.androidbox.data.worker_manager
+
+import android.content.Context
+import androidx.hilt.work.HiltWorker
+import androidx.work.CoroutineWorker
+import androidx.work.WorkerParameters
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedInject
+import kotlinx.coroutines.Deferred
+import kotlinx.coroutines.async
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.supervisorScope
+import me.androidbox.data.local.dao.TaskDao
+import me.androidbox.data.local.entity.TaskEntity
+import me.androidbox.data.mapper.toTaskDto
+import me.androidbox.data.remote.network.task.TaskService
+import me.androidbox.data.remote.util.CheckResult.checkResult
+import me.androidbox.data.worker_manager.constant.Constant.RETRY_COUNT
+import me.androidbox.domain.constant.SyncAgendaType
+
+@HiltWorker
```

Comment:
You are going to have one worker for each agenda type?

Personally I just have 1 worker, that syncs everything

Theres not really a benefit in having 3 workers over just 1